### DISCUSSION TITLE
Clear multiple ui.input & ui.select elements at once?

### DISCUSSION THREAD

**Discussion by pythonormen** [zauberzeug/nicegui#467] [Q&A] [answered]
I have a ui.card with multiple  ui.input & ui.select elements under it. Is there any way to have a button to clear/reset everything within the card? Right now I'm addressing every element individually.

**Comment by AL-THE-BOT-FATHER** [comment]
I'm pretty sure you have to address every element individually. You could create a function for the button click and loop through the list of the UI elements you want to clear. If you post the code, we can help out better. Cheers.

**Comment by falkoschindler** [comment, marked as answer]
Currently there is no official reset method. But you can build one yourself, e.g. setting all value elements to `None`:

```py
from nicegui import ui
from nicegui.elements.mixins.value_element import ValueElement

with ui.column() as column:
    ui.label('Please, fill in the form:')
    ui.input('Name', value='John Doe')
    ui.number('Age', value=18)
    ui.label('Height:')
    ui.slider(value=1.8, min=1.0, max=2.0, step=0.01)

    def reset():
        for child in column.default_slot.children:
            if isinstance(child, ValueElement):
                child.value = None
    ui.button('Reset', on_click=reset)

ui.run()
```

**Reply by AL-THE-BOT-FATHER** [reply]
This is awesome. ^^^^
